myapp/scheduler/stock_api_update.py

```diff
@@ -24,3 +32,26 @@ def stock_api_update():
             index += 1
     except KeyError as e:
         pass
+
+
+def update_existing_stocks():
+    s = WatchStock.objects.all().values("stock_id").distinct()
+    for w in s:
+        stock = Stock.objects.all().filter(symbol=w.get('stock_id'))[0]
+        if (datetime.now(stock.last_modified.tzinfo) - stock.last_modified).total_seconds() > 5:
+            data = stock_api.get_stock_info(stock.symbol)
+            Stock.objects.filter(symbol=stock.symbol).update(
```

Comment:
i'm not sure why you searching for the stock again, why not just `stock.update(...`